Seminars/Sem.02/Pract.02/Task07.cpp

```diff
@@ -0,0 +1,174 @@
+
+    User user;
+
+    std::copy(&name[0],&name[sizeof user.name],user.name);
+    std::copy(&email[0],&email[sizeof user.email],user.email);
+    std::copy(&password[0],&password[sizeof user.password],user.password);
+
+    return user;
+
+}
+
+bool downLoadData(System &system){
+
+    std::ifstream file(SYSTEM_PATH,std::ios::in);
+    if(!file.is_open())return false;
+
+    char name[MAX_SIZE];
+    char email[MAX_SIZE];
+    char password[MAX_SIZE];
+
+
+    while(!file.eof()){
+
+        file>>name;
+        file>>email;
+        file>>password;
+        if(name[0] != '\0' or email[0] != '\0' or password[0] != '\0')
+            addUserToSystem(createUser(name,email,password),system);
+
+    }
+
+    file.close();
+
+    return true;
+
+}
+
+bool saveData(System &system){
+
+    std::ofstream file(SYSTEM_PATH,std::ios::out);
+    file.clear();
+    for(int i = 0; i < system.size;i++){
+
+        file<<system.userList[i].name<<std::endl;
+        file<<system.userList[i].email<<std::endl;
+        if(system.size - 1 == i) file<<system.userList[i].password;
+        else file<<system.userList[i].password<<std::endl;
+
+    }
+
+    file.close();
+
+}
+
+void printUser(const User user){
+
+    std::cout<<"Name = "<<user.name<<" ";
+    std::cout<<"Email = "<<user.email<<" ";
+    std::cout<<"Password = "<<user.password<<"\n";
+
+}
+
+void printSystem(const System system){
+
+    for(int i = 0; i < system.size;i++){
+        printUser(system.userList[i]);
+    }
+
+}
+bool compareStr(char left[],char right[]){
+
+    for(int i = 0;left[i]!='\0' or right[i]!='\0';i++){
+        if(left[i] != right[i])return false;
+    }
+    return true;
+}
+bool loginUser(User user,System &system){
+
+    for(int i=0;i< system.size;i++){
+
+        if( compareStr(system.userList[i].email,user.email)
+            and
+            compareStr(system.userList[i].password,user.password) )
+            return true;
+
+    }
+
+    return false;
+
+}
+
+int main(){
+
+    System system;
+    downLoadData(system);
+
+    char registerCommand[] = "register";
+    char loginCommand[] = "login";
+    char command[8];
+
+    char name[MAX_SIZE];
+    char email[MAX_SIZE];
+    char password[MAX_SIZE];
+
+    std::cout<<"Enter Command =>  ";
+    std::cin>>command;
+
+    if(compareStr(command,registerCommand)){
+
+        std::cout<<"Name =>  ";
+        std::cin>>name;
+        std::cout<<"Email =>  ";
+        std::cin>>email;
+        std::cout<<"Password =>  ";
+        std::cin>>password;
+
+        addUserToSystem(createUser(name,email,password),system);
+
+    }
+    else if(compareStr(command,loginCommand)){
+
+        std::cout<<"Email =>  ";
+        std::cin>>email;
+        std::cout<<"Password =>  ";
+        std::cin>>password;
+
+        if(loginUser(createUser("",email,password),system))
+            std::cout<<"Login successful";
+        else std::cout<<"No matches";
+    }
+    else std::cout<<"Wrong Command";
+
+    saveData(system);
+
```

Comment:
What happens with the dynamic memory for userList in system

Reply:
in saceData() now